controllers/operandrequest/reconcile_operator.go

```diff
@@ -213,10 +213,19 @@ func (r *Reconciler) deleteSubscription(ctx context.Context, operandName string,
 
 	namespace := r.GetOperatorNamespace(op.InstallMode, op.Namespace)
 	sub, err := r.GetSubscription(ctx, operandName, namespace, op.PackageName)
-
+	originalsub := sub.DeepCopy()
 	if apierrors.IsNotFound(err) {
 		klog.V(3).Infof("There is no Subscription %s or %s in the namespace %s", operandName, op.PackageName, namespace)
 		return nil
+	} else if err != nil {
+		klog.Errorf("Failed to get Subscription %s or %s in the namespace %s", operandName, op.PackageName, namespace)
+		return err
+	}
+
+	if sub == nil || sub.Labels == nil {
```

Comment:
If the label is not existing, indicating it is not managed by ODLM.
If `sub` is `nil`, does it mean that the subscription is not exists and `apierrors.IsNotFound(err)` is `true`?

Reply:
Yes, I can modify it to `if sub.Labels == nil {`